pkgs/games/xonotic/default.nix

```diff
@@ -98,32 +101,46 @@ let
 
     # Xonotic needs to find libcurl.so at runtime for map downloads
     dontPatchELF = true;
-    postFixup = lib.optionalString withDedicated ''
-      patchelf --add-needed ${curl.out}/lib/libcurl.so $out/bin/xonotic-dedicated
-    '' + lib.optionalString withGLX ''
-      patchelf \
-          --add-needed ${curl.out}/lib/libcurl.so \
-          --add-needed ${libvorbis}/lib/libvorbisfile.so \
-          --add-needed ${libvorbis}/lib/libvorbis.so \
-          --add-needed ${libGL.out}/lib/libGL.so \
-          $out/bin/xonotic-glx
-    '' + lib.optionalString withSDL ''
-      patchelf \
+
+    commonPatchelfArgs = ''
           --add-needed ${curl.out}/lib/libcurl.so \
-          --add-needed ${libvorbis}/lib/libvorbisfile.so \
-          --add-needed ${libvorbis}/lib/libvorbis.so \
-          $out/bin/xonotic-sdl
+          --add-needed ${zlib.out}/lib/libz.so \
+          --add-needed $out/lib/libd0_blind_id.so \
     '';
+
+    commonClientPatchelfArgs= ''
+          --add-needed ${libvorbis.out}/lib/libvorbis.so \
+          --add-needed ${libvorbis.out}/lib/libvorbisfile.so \
+          --add-needed ${libvorbis.out}/lib/libvorbisenc.so \
+          --add-needed ${libtheora.out}/lib/libtheora.so \
+          --add-needed ${libtheora.out}/lib/libtheoraenc.so \
+          --add-needed ${libtheora.out}/lib/libtheoradec.so \
+          --add-needed ${libpng.out}/lib/libpng16.so \
+          --add-needed ${freetype.out}/lib/libfreetype.so \
+    '';
+
+    postFixup =
+      lib.optionalString withDedicated (''
+          patchelf '' + commonPatchelfArgs + ''
+          "$out/bin/xonotic-dedicated"
+      '') + lib.optionalString withGLX (''
+          patchelf '' + commonPatchelfArgs + commonClientPatchelfArgs + ''
+          --add-needed ${libGL.out}/lib/libGL.so \
+          "$out/bin/xonotic-glx"
+      '') + lib.optionalString withSDL (''
+          patchelf '' + commonPatchelfArgs + commonClientPatchelfArgs + ''
+          "$out/bin/xonotic-sdl"
+      '');
   };
 
 in rec {
   xonotic-data = fetchzip {
     name = "xonotic-data-${version}";
     url = "https://dl.xonotic.org/${name}.zip";
-    sha256 = "1ygkh0v68y4sd1w5vpk8dgb65h5jm599hwszdfgjp3ax4d3ml81x";
+    sha256 = "15caj11v9hhr7w55w3rs1rspblzr9lg1crqivbn9pyyq0rif8cpl";
     extraPostFetch = ''
       cd $out
-      rm -rf $(ls | grep -v "^data$")
+      rm -rf $(ls | grep -v "^data$\|^key_0.d0pk$")
```

Comment:
What is the purpose of this change?

Reply:
I think that's the public key of the UID tracking infrastructure. This change does keep it from getting deleted in the data directory. IIRC this change was necessary for it to work. Possibly I'm conflating this with the required d0_blind_id building, since I was poking the thing from all directions back then in june ;)

Reply:
Thanks for the explanation